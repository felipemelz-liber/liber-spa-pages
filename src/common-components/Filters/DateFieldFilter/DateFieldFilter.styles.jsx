import styled from 'styled-components';

export const Box = styled.div`
  & > div > div {
    border-right: 0px;
  }
`;

export const Text = styled.p`
  font-family: Roboto;
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.1px;
  text-align: left;
  color: #405f71;
  margin: 0;
  padding: 18px 32px 0px 32px;
`;
