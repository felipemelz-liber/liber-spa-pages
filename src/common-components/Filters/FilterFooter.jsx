import React from 'react';
import PropTypes from 'prop-types';
import { Footer, FilterButton, ClearButton } from './Filters.styles';

const FilterFooter = ({ disabled, onFilter, onClear }) => {
  return (
    <Footer>
      <FilterButton disabled={disabled} onClick={onFilter}>
        FILTRAR
      </FilterButton>
      <ClearButton onClick={onClear} disabled={disabled}>
        LIMPAR
      </ClearButton>
    </Footer>
  );
};

FilterFooter.propTypes = {
  disabled: PropTypes.bool,
  onFilter: PropTypes.func,
  onClear: PropTypes.func,
};

FilterFooter.defaultProps = {
  disabled: false,
  onFilter: () => null,
  onClear: () => null,
};

export default FilterFooter;
