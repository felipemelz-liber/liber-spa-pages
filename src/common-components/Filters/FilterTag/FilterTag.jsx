import React from 'react';
import PropTypes from 'prop-types';
import {
  FilterTag as Tag,
  SelectedLabelContainer,
  SelectedLabelOption,
  SelectedSwitch,
} from './FilterTag.styles';

const isSelected = list => {
  if (list) {
    return list.filter(item => Boolean(item)).length !== 0;
  }
  return false;
};

const format = list => {
  return list.join(' ');
};

const FilterTag = ({ selected, formatSelected, label, onClear, show, ...props }) => {
  const [hover, setHover] = React.useState(false);
  let tagProps;
  if (isSelected(selected)) {
    const nodeLabel = (
      <SelectedLabelContainer>
        <SelectedSwitch switch={hover || show}>
          <SelectedLabelOption>{formatSelected(selected)}</SelectedLabelOption>
          <SelectedLabelOption>{label}</SelectedLabelOption>
        </SelectedSwitch>
      </SelectedLabelContainer>
    );
    tagProps = {
      label: nodeLabel,
      closable: true,
      onClose: onClear,
      onMouseEnter: () => setHover(true),
      onMouseLeave: () => setHover(false),
      gray: !show,
      show,
      ...props,
    };
  } else {
    tagProps = { label, show, ...props };
  }
  return <Tag {...tagProps} />;
};

FilterTag.propTypes = {
  selected: PropTypes.arrayOf(PropTypes.any),
  formatSelected: PropTypes.func,
  onClear: PropTypes.func,
  label: PropTypes.string,
  show: PropTypes.bool,
};

FilterTag.defaultProps = {
  selected: [],
  formatSelected: format,
  onClear: () => {},
  show: false,
  label: '',
};

export default FilterTag;
