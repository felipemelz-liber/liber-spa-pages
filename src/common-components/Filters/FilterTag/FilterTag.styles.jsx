import styled, { css } from 'styled-components';
import { ModalTag } from 'liber-components';

const isGray = ({ gray }) => {
  if (gray)
    return css`
      background-color: #f4f5f7;
      border-color: #f4f5f7;
    `;
  return null;
};

export const FilterTag = styled(ModalTag)`
  font-size: 16px;
  min-height: 36px;
  border-radius: 4px;
  font-weight: 400;
  ${isGray}
`;

export const SelectedLabelContainer = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  overflow: hidden;
`;

export const SelectedSwitch = styled.div`
  height: 52px;
  width: 100%;
  padding-top: 6px;
  display: flex;
  flex-direction: column;
  transform: ${props => (props.switch ? 'translateY(-26px)' : 'translateY(0px)')};
  transition: transform 250ms ease-in-out;
`;

export const SelectedLabelOption = styled.span`
  width: 100%;
  height: 26px;
  display: flex;
  justify-content: center;
  align-items: center;
`;
