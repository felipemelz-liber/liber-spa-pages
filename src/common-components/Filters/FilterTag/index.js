export { default } from './FilterTag.jsx';
