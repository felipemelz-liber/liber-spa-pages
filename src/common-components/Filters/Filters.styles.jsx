import styled, { css } from 'styled-components';
import { ModalTag, CardFooter, Button, handleThemeFromObject, TextField } from 'liber-components';

const getTheme = (key, fallback) =>
  css`
    ${({ theme }) => handleThemeFromObject(theme, key, fallback)}
  `;

export const Container = styled.div`
  min-height: 68px;
  width: 100%;
  border-right: 1px solid #cdd7df;
  border-left: 1px solid #cdd7df;
  padding: 0px 16px;
  display: flex;
  justify-content: flex-start;
  align-items: center;
`;

export const GreyFilterTag = styled(ModalTag)`
  font-weight: 400;
`;

export const Label = styled.div`
  font-family: Roboto;
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.1px;
  text-align: left;
  color: #405f71;
  margin-right: 16px;
`;

export const Footer = styled(CardFooter)`
  width: 100%;
  min-height: 52px;
  height: 52px;
  background-color: #ffffff;
  border-radius: 0px 0px 4px 4px;
  padding-top: unset;
  padding-bottom: unset;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex-direction: row-reverse;
`;

export const ClearButton = styled(Button).attrs(() => ({ version: '2' }))`
  font-weight: 400;
`;

export const FilterButton = styled(Button).attrs(() => ({ version: '2' }))`
  font-weight: 400;
  color: ${getTheme('colors.liber', '#009dff')};
`;

export const Field = styled(TextField)``;

export const CleanAll = styled(FilterButton)`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.1px;
`;

export const Filtered = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  line-height: 22px;
  letter-spacing: 0.1px;
  text-align: right;
  color: #405f71;
  margin-left: auto;
`;
