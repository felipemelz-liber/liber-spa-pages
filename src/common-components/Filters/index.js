export { default as ArrayFieldFilter } from './ArrayFieldFilter/ArrayFieldFilter';
export { default as TextFieldFilter } from './TextFieldFilter/TextFieldFilter';
export { default as DateFieldFilter } from './DateFieldFilter/DateFieldFilter';
