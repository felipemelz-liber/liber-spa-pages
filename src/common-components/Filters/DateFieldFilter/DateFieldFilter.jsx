import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Range } from 'liber-components/components/Inputs';
import moment from 'moment';
import { Box, Text } from './DateFieldFilter.styles.jsx';
import FilterTag from '../FilterTag';
import FilterFooter from '../FilterFooter';

const hasClearedFilters = selectedItems =>
  selectedItems.filter(filter => filter !== null).length === 0;

const DateFieldFilter = ({ label, description, onFilter, onClear, selectedItems }) => {
  const [showTag, setShowTag] = useState(false);
  const [rangeField, setRangeField] = useState({ start: null, end: null });
  const [displayDate, setDisplayDate] = useState(rangeField.start || moment());

  useEffect(() => {
    if (hasClearedFilters(selectedItems)) {
      setRangeField({ start: null, end: null });
    }
  }, [selectedItems]);

  const isDisabled = () => !(rangeField.start || rangeField.end);
  const handleClear = () => {
    setRangeField({ start: null, end: null });
    onClear();
  };
  const handleFilter = () => {
    onFilter(rangeField);
    setShowTag(false);
  };
  const toggleShow = (shouldShow = undefined) =>
    shouldShow === undefined ? setShowTag(!showTag) : setShowTag(shouldShow);

  const formatLabel = selected => {
    const [start, end, singleDay] = selected;
    if (start && end) {
      return `${moment(start).format('DD/MM/YYYY')} a ${moment(end).format('DD/MM/YYYY')}`;
    }
    if (singleDay) {
      return moment(singleDay).format('DD/MM/YYYY');
    }
    if (start) {
      return moment(start).format('DD/MM/YYYY');
    }
    if (end) {
      return moment(end).format('DD/MM/YYYY');
    }
    return 'Data de Criação';
  };
  const navigate = date => setDisplayDate(date);

  return (
    <FilterTag
      show={showTag}
      onChange={toggleShow}
      onClear={handleClear}
      selected={selectedItems}
      formatSelected={formatLabel}
      label={label}
      overflow="center"
    >
      <Box>
        <Text>{description}</Text>
        <Range
          showTime={false}
          onChangeDates={(start, end) => setRangeField({ start, end })}
          startDate={rangeField.start}
          endDate={rangeField.end}
          displayDate={displayDate}
          onNavigate={navigate}
        />
      </Box>
      <FilterFooter disabled={isDisabled()} onFilter={handleFilter} onClear={handleClear} />
    </FilterTag>
  );
};

DateFieldFilter.propTypes = {
  label: PropTypes.string,
  description: PropTypes.string,
  onFilter: PropTypes.func,
  onClear: PropTypes.func,
  selectedItems: PropTypes.arrayOf(PropTypes.string),
};

DateFieldFilter.defaultProps = {
  label: 'Data',
  description: 'Selecione uma data de início e uma data de fim para o período desejado:',
  onFilter: () => null,
  onClear: () => null,
  selectedItems: [],
};

export default DateFieldFilter;
