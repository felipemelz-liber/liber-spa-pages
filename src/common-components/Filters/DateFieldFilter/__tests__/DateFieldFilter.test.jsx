import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import 'jest-styled-components';
import DateFieldFilter from '../DateFieldFilter';
import { FilterButton, ClearButton } from '../../Filters.styles';
import FilterFooter from '../../FilterFooter';
import FilterTag from '../../FilterTag';

describe('Testing DateFieldFilter Component', () => {
  configure({ adapter: new Adapter() });

  const props = {
    label: 'Date label',
    description: 'Selecione o perÃ­odo:',
    onFilter: sinon.spy(),
    onClear: sinon.spy(),
  };

  let clock;
  beforeEach(() => {
    const now = new Date('1234-01-01');
    clock = sinon.useFakeTimers(now.getTime());
  });
  afterEach(() => {
    clock.restore();
  });

  it('should render correctly without props', () => {
    const wrapper = shallow(<DateFieldFilter />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with props', () => {
    const wrapper = shallow(<DateFieldFilter {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call onFilter when clicking on FilterButton', () => {
    const wrapper = shallow(<DateFieldFilter {...props} />);
    const filterFooter = wrapper.find(FilterFooter);
    const filterButton = filterFooter.dive().find(FilterButton);
    const { onFilter } = props;
    filterButton.simulate('click');
    expect(onFilter.called).toBeTruthy();
  });
  it('should call onClear when clicking on ClearButton', () => {
    const wrapper = shallow(<DateFieldFilter {...props} />);
    const filterFooter = wrapper.find(FilterFooter);
    const clearButton = filterFooter.dive().find(ClearButton);
    const { onClear } = props;
    clearButton.simulate('click');
    expect(onClear.called).toBeTruthy();
  });
  it('should call onClear when FilterTag clear event is fired', () => {
    const wrapper = shallow(<DateFieldFilter {...props} />);
    const tag = wrapper.find(FilterTag);
    const { onClear } = props;
    tag.simulate('clear');
    expect(onClear.called).toBeTruthy();
  });
});
