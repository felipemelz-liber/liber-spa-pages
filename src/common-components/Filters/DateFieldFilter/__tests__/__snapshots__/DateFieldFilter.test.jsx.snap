// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DateFieldFilter Component should render correctly with props 1`] = `
<FilterTag
  formatSelected={[Function]}
  label="Date label"
  onChange={[Function]}
  onClear={[Function]}
  overflow="center"
  selected={Array []}
  show={false}
>
  <styled.div>
    <styled.p>
      Selecione o período:
    </styled.p>
    <Range
      displayDate={"1234-01-01T00:00:00.000Z"}
      endDate={null}
      onChangeDates={[Function]}
      onNavigate={[Function]}
      showTime={false}
      startDate={null}
    />
  </styled.div>
  <FilterFooter
    disabled={true}
    onClear={[Function]}
    onFilter={[Function]}
  />
</FilterTag>
`;

exports[`Testing DateFieldFilter Component should render correctly without props 1`] = `
<FilterTag
  formatSelected={[Function]}
  label="Data"
  onChange={[Function]}
  onClear={[Function]}
  overflow="center"
  selected={Array []}
  show={false}
>
  <styled.div>
    <styled.p>
      Selecione uma data de início e uma data de fim para o período desejado:
    </styled.p>
    <Range
      displayDate={"1234-01-01T00:00:00.000Z"}
      endDate={null}
      onChangeDates={[Function]}
      onNavigate={[Function]}
      showTime={false}
      startDate={null}
    />
  </styled.div>
  <FilterFooter
    disabled={true}
    onClear={[Function]}
    onFilter={[Function]}
  />
</FilterTag>
`;
